### PR TITLE
haywood/ch14955/add-test-for-deeply-nested-fragments, add shadow for …

### PR DESCRIPTION
**Pull request by haywoood**
Also add shadow for an optional way to run tests during development. Gives a browser repl and nicer test runner, useful for development.

### CODE REVIEW

File: src/artemis/document.cljc [hunk 1]
```diff
@@ -61,8 +61,8 @@
 (defn compose
   "Compose multiple parsed documents together to create a single document."
   {:added "0.1.0"}
-  [doc & docs]
-  (apply merge-with into doc docs))
+  [root-doc & docs]
```
Comment (haywoood) [inline src/artemis/document.cljc]: This was the crux of my witch hunt. I had composed multiple fragments, but we treat the first fragment as the main entry point, and my 'main fragment' was positioned last. So it seemed like deeply nested fragments were not working, when if you re-order the position the problem is solved. Changed the name of the first argument to make that mechanic more clear.

File: test/artemis/core_test.cljs [hunk 2]
```diff
@@ -23,32 +23,32 @@
...
+                 (testing "calls the callback"
+                   (let [_ (<! result-chan) ; first local result
+                         _ (<! result-chan) ; first remote result
+                         ]
+                     (is (= @counter 1)))
+                   (let [_ (<! result-chan) ; second local result
+                         _ (<! result-chan) ; second remote result
+                         ]
+                     (is (= @counter 2))))
+                 (testing "unwatch"
```
Comment (haywoood) [inline test/artemis/core_test.cljs]: Correcting spelling, the rest is indentation

File: test/artemis/document_test.cljs [hunk 3]
```diff
@@ -8,11 +8,6 @@
   (let [doc (d/parse-document "{ a b c }")]
     (is (some? doc))))
 
-(deftest parse-document-files
```
Comment (yedi) [inline test/artemis/document_test.cljs]: Why was this test removed

**Comment by haywoood** [comment]
@yedi yea that's all untouched, the shadow dev route is 100% opt-in

**Pull request merged**